confgenerator/config.go

```diff
@@ -149,6 +153,24 @@ func newValidator() *validator.Validate {
 		}
 		return t >= tmin
 	})
+	// time_denomination validates that the duration is whole to some time denomination
+	// eg. s, ms, m, h
+	v.RegisterValidation("whole_time_denomination", func(fl validator.FieldLevel) bool {
+		t, ok := fl.Field().Interface().(time.Duration)
+		if !ok {
+			panic(fmt.Sprintf("whole_time_denomination: could not convert %s to time duration", fl.Field().String()))
+		}
+
+		var denomination time.Duration
+		switch fl.Param() {
+		case "second":
```

Comment:
Why not just have the argument be "1s" and parse it as a duration?

Reply:
I combined this idea and `multipleof` into a solution that should be more scalable than this original implementation.

Reply:
Looks like this one has been resolved. Thanks!